File: backend/src/Enum/FeatureType.php
<?php

declare(strict_types=1);

namespace App\Enum;

enum FeatureType: string
{
    case TransitStation = 'transit_station';
    case CommercialAirport = 'commercial_airport';
    case RailStation = 'rail_station';
    case MetroLine = 'metro_line';
    case HighSpeedRailLine = 'high_speed_rail_line';
    case Mountain = 'mountain';
    case Park = 'park';
    case BodyOfWater = 'body_of_water';
    case Coastline = 'coastline';
    case AdminBoundary1st = 'admin_boundary_1st';
    case AdminBoundary2nd = 'admin_boundary_2nd';
    case AdminBoundary3rd = 'admin_boundary_3rd';
    case AdminBoundary4th = 'admin_boundary_4th';
    case BorderInternational = 'border_international';
    case Border1st = 'border_admin_1st';
    case Border2nd = 'border_admin_2nd';
    case Hospital = 'hospital';
    case Library = 'library';
    case Museum = 'museum';
    case MovieTheater = 'movie_theater';
    case Zoo = 'zoo';
    case Aquarium = 'aquarium';
    case AmusementPark = 'amusement_park';
    case GolfCourse = 'golf_course';
    case Consulate = 'consulate';

    public function label(): string
    {
        return match ($this) {
            self::TransitStation => 'transit station',
            self::CommercialAirport => 'airport',
            self::RailStation => 'rail station',
            self::MetroLine => 'metro line',
            self::HighSpeedRailLine => 'high-speed rail',
            self::Mountain => 'mountain',
            self::Park => 'park',
            self::BodyOfWater => 'body of water',
            self::Coastline => 'coastline',
            self::AdminBoundary1st => '1st admin division',
            self::AdminBoundary2nd => '2nd admin division',
            self::AdminBoundary3rd => '3rd admin division',
            self::AdminBoundary4th => '4th admin division',
            self::BorderInternational => 'international border',
            self::Border1st => '1st division border',
            self::Border2nd => '2nd division border',
            self::Hospital => 'hospital',
            self::Library => 'library',
            self::Museum => 'museum',
            self::MovieTheater => 'movie theater',
            self::Zoo => 'zoo',
            self::Aquarium => 'aquarium',
            self::AmusementPark => 'amusement park',
            self::GolfCourse => 'golf course',
            self::Consulate => 'consulate',
        };
    }

    public function geometryKind(): string
    {
        return match ($this) {
            self::CommercialAirport,
            self::Park,
            self::BodyOfWater,
            self::AdminBoundary1st,
            self::AdminBoundary2nd,
            self::AdminBoundary3rd,
            self::AdminBoundary4th => 'areal',
            self::MetroLine,
            self::HighSpeedRailLine,
            self::Coastline,
            self::BorderInternational,
            self::Border1st,
            self::Border2nd => 'linear',
            default => 'point',
        };
    }

    public function adminRank(): ?int
    {
        return match ($this) {
            self::AdminBoundary1st => 1,
            self::AdminBoundary2nd => 2,
            self::AdminBoundary3rd => 3,
            self::AdminBoundary4th => 4,
            default => null,
        };
    }

    public static function adminTypeForRank(int $rank): ?FeatureType
    {
        return match ($rank) {
            1 => self::AdminBoundary1st,
            2 => self::AdminBoundary2nd,
            3 => self::AdminBoundary3rd,
            4 => self::AdminBoundary4th,
            default => null,
        };
    }

    public function isAdminBorder(): bool
    {
        return match ($this) {
            self::BorderInternational, self::Border1st, self::Border2nd => true,
            default => false,
        };
    }
}

File: android/app/src/main/java/fr/gshz/hideandseek/domain/model/FeatureType.kt
package fr.gshz.hideandseek.domain.model

import androidx.annotation.StringRes
import fr.gshz.hideandseek.R

/**
 * POI categories a Matching/Measuring/Tentacles question can target. [minGameSize] and the
 * per-category availability follow the game rules; the UI filters by both.
 */
enum class FeatureType(val wireValue: String, @get:StringRes val labelRes: Int) {
    TransitStation("transit_station", R.string.feature_type_transit_station),
    CommercialAirport("commercial_airport", R.string.feature_type_commercial_airport),
    RailStation("rail_station", R.string.feature_type_rail_station),
    MetroLine("metro_line", R.string.feature_type_metro_line),
    HighSpeedRailLine("high_speed_rail_line", R.string.feature_type_high_speed_rail_line),
    Mountain("mountain", R.string.feature_type_mountain),
    Park("park", R.string.feature_type_park),
    BodyOfWater("body_of_water", R.string.feature_type_body_of_water),
    Coastline("coastline", R.string.feature_type_coastline),
    BorderInternational("border_international", R.string.feature_type_border_international),
    Border1st("border_admin_1st", R.string.feature_type_border_admin_1st),
    Border2nd("border_admin_2nd", R.string.feature_type_border_admin_2nd),
    AdminBoundary1st("admin_boundary_1st", R.string.feature_type_admin_boundary_1st),
    AdminBoundary2nd("admin_boundary_2nd", R.string.feature_type_admin_boundary_2nd),
    AdminBoundary3rd("admin_boundary_3rd", R.string.feature_type_admin_boundary_3rd),
    AdminBoundary4th("admin_boundary_4th", R.string.feature_type_admin_boundary_4th),
    Hospital("hospital", R.string.feature_type_hospital),
    Library("library", R.string.feature_type_library),
    Museum("museum", R.string.feature_type_museum),
    MovieTheater("movie_theater", R.string.feature_type_movie_theater),
    Zoo("zoo", R.string.feature_type_zoo),
    Aquarium("aquarium", R.string.feature_type_aquarium),
    AmusementPark("amusement_park", R.string.feature_type_amusement_park),
    GolfCourse("golf_course", R.string.feature_type_golf_course),
    Consulate("consulate", R.string.feature_type_consulate),
    ;

    companion object {
        fun fromWireValue(value: String): FeatureType? = entries.firstOrNull { it.wireValue == value }
    }
}
